ci/lib/cacheable_install_functions.inc.sh

```diff
@@ -210,6 +210,10 @@ install_gpg() {
         _install_gpg component-git-ref 2501a48 f73605e      d9c4183   909133b   3df0cd3 0e2e53c  c6702d7
         # _install_gpg component-git-ref 7e45b50 c66594d      cf88dca   57cf9d6   4243085 6e8ad31  d4e5979
         ;;
+      "2.3.1")
+        #                              npth libgpg-error       libgcrypt libassuan libksba pinentry gnupg
+        _install_gpg component-git-ref 1.6  libgpg-error-1.42  1.9.3     2.5.5    1.6.0   1.1.1    2.3.1
```

Comment:
@ribose-jeffreylau Do you have any idea why it doesn't work with 1.42 but works with libgpg-error-1.42? Is it something changed on GnuPG repository or in gpg build scripts? The same with libgcrypt's vesion.

Reply:
@ribose-jeffreylau Okay, that was my mistake. Now the question is how to correctly update automake to 1.16.3 for centos8, as it is required since GnuPG 2.3.0 release?

Reply:
@ni4 Thanks!  I've opened this issue to investigate: https://github.com/rnpgp/gpg-build-scripts/issues/59

Reply:
A more related PR is #1607 .  Once that is merged, we can try to rebase this.